### PR TITLE
Fixed rx observing for the connection state.

### PR DESCRIPTION
**Pull request by buh**
For rx subscriptions important to start with an initial connection state.
I made `WebSocket.connectionState` as Atomic and shared it on `Client`.
Also, I added current values for Client unread count and user on subscribe.

### CODE REVIEW

File: Sources/Client/Client/Client+Events.swift [hunk 1]
```diff
@@ -118,6 +118,11 @@ extension Client {
             }
         }
         
+        // Return the current user.
+        if !user.isUnknown {
+            callback(user)
+        }
+
```
Comment (b-onc) [inline Sources/Client/Client/Client+Events.swift:125]: This is repeating lines 116-118 why?

File: Sources/Client/Client/Client+Events.swift [hunk 2]
```diff
@@ -136,6 +141,11 @@ extension Client {
             callback(self.unreadCount)
         }
         
+        // Return the current unread count.
+        if !user.isUnknown {
+            callback(unreadCount)
+        }
+
```
Comment (b-onc) [inline Sources/Client/Client/Client+Events.swift:148]: This is repeating lines 140-141, why?

File: Sources/Client/Client/Client+Events.swift (hunk 1 above)
Comment (buh) [reply Sources/Client/Client/Client+Events.swift:125]: It's a different scope.
In 116-118 the callback will be call on event, the new lines will send the current value immediately.
The same for the Client unread count.

Comment (b-onc) [reply Sources/Client/Client/Client+Events.swift:125]: Nope, lines 116-118 will run immediately in `eventsHandlingQueue`, not on event. If it's not running, that's a bug and this is not the fix

Comment (buh) [reply Sources/Client/Client/Client+Events.swift:125]: Why do you think it's a bug?

Comment (b-onc) [reply Sources/Client/Client/Client+Events.swift:125]: Because lines 116-118 do not run when an event is fired. They run immediately in async manner.

**Comment by b-onc** [comment]
Should fix #245, need testing

**Pull request merged**